pkg/odo/cli/catalog/list/components.go

```diff
@@ -57,10 +57,14 @@ func (o *ListComponentsOptions) Complete(name string, cmd *cobra.Command, args [
 	}
 
 	if experimental.IsExperimentalModeEnabled() {
-		o.catalogDevfileList, err = catalog.ListDevfileComponents()
+		o.catalogDevfileList, err = catalog.ListDevfileComponents("")
 		if err != nil {
 			return err
 		}
+
+		if o.catalogDevfileList.DevfileRegistries == nil {
+			log.Warning("Please run 'odo registry add <registry name> <registry URL>' to add registry for listing devfile components\n")
```

Comment:
The reason is `odo catalog list components` list both devfile component and s2i component, we don't want the `error` blocks user from viewing the s2i components, so we use `warn` instead.

The use case we cover is that user is on experimental mode without specifying any registry, but user only wants to list/create s2i component, we should support that and only warns user in case user wants to list/create devfile component later.